ml_package_framework/data_processing/data_processors.py

```diff
@@ -77,7 +78,262 @@ def load_data(self, path: str, *args, **kwargs) -> pd.DataFrame:
+        dataframe.
+
+        Parameters
+        ----------
+        train_df : pd.DataFrame
+            The training dataframe, which includes a unique 'id' for each row.
+        valid_melt : pd.DataFrame
+            The validation dataframe, containing at least the 'target' column.
+        submission_df : pd.DataFrame
+            A dataframe used for filtering, containing unique 'id' values.
+
+        Returns
+        -------
+        pd.DataFrame
+            The filtered and merged training dataframe, now containing 'target' values
+            and rows corresponding to the 'id's in the submission dataframe.
+        """
+        train_df["target"] = valid_melt["target"].astype(int)
+        train_df = train_df[train_df.id.isin(submission_df.id.unique())].reset_index(
+            drop=True
+        )
+        return train_df
+
+    def calculate_total_ordered_(self, train_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Calculate the total quantity ordered for each category in the training dataset.
+
+        This method computes a new column 'total_ordered' in the training dataframe. It
+        represents the total quantity of orders for each category across all users. This
+        is achieved by summing up the 'ordered' column for each 'category' and mapping
+        these sums back to the corresponding categories in the training dataset.
+
+        Parameters
+        ----------
+        train_df : pd.DataFrame
+            The training dataframe, containing at least 'category' and 'ordered' columns.
+
+        Returns
+        -------
+        pd.DataFrame
+            The modified training dataframe, now including a 'total_ordered' column
+            indicating the total quantity ordered for each category.
+        """
+        total_ordered = train_df.groupby("category")["ordered"].sum()
+        train_df["total_ordered"] = train_df["category"].map(total_ordered)
+        return train_df
+
+    def process_data(self, data: pd.DataFrame, *args, **kwargs) -> pd.DataFrame:
+        """
+        Process raw data through various transformation steps
+        to prepare it for analysis or modeling.
+
+        This method orchestrates a series of data transformation steps on the input raw dataframe.
+        It starts by creating dummy variables and grouping the data, then adds a purchase counter,
+        splits the dataset into training and validation sets, melts these sets into a long format,
+        calculates additional features, merges and filters the data based on a submission dataframe,
+        and finally calculates the total quantity ordered for each category. The method is flexible
+        to accept additional arguments and keyword arguments, which can be used in future
+        enhancements or for passing additional parameters to the underlying methods.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            The raw input dataframe to be processed.
+        sub : pd.DataFrame
```

Comment:
you have a mismatch between paramateres mentioned here and an acrtual method signature. You dont pass sub param, but unpack it from kwargs.

I think you dont need args, kwargs here. If you dont expect other inputs, lets use data and sub as patameters only.

But more general questions - can we think of a more abstract name rather than sub? Sub is about kaggle sumbission while we are building a more general thing.